site.yml

```diff
@@ -2,7 +2,11 @@
 - name: common role for all hosts
   hosts: all:!vyatta-*
   any_errors_fatal: true
-  gather_facts: force
+  pre_tasks:
+    - setup:
+      delegate_to: "{{ item }}"
+      delegate_facts: True
+      with_items: "{{ groups['all'] }}"
```

Comment:
Instead of having a pre_task for every role that had forced gathered facts, can you do a top level play that has this task in it, with `tags: always`?

This will mean we get the fact gathering even if we do say `--tags nova` in execution, AND it means we gather the facts once and utilize the in-memory fact caching, rather than running a serialized setup task in each and every play (which would be a big slow down).